tests/conftest.py

```diff
@@ -559,9 +560,22 @@ def identity(x):
     return x
 
 
-def fieldpath_test_mode(func):
-    def wrapper(*args, **kwargs):
-        FieldPath.__test_mode = True
-        func(*args, **kwargs)
+@contextmanager
+def fieldpath_test_mode():
+    """Temporarily turns on `FieldPath`'s testing mode which undo's custom DSL-type
+      behavior that overrides `__eq__`.
 
-    return wrapper
+    `FieldPath.__test_mode` is the name of the field inside the class but here, we use
```

Comment:
Wow! I did not know you could do that

Reply:
Name mangling is def. an interesting feature in Python haha